clippy_lints/src/types.rs

```diff
@@ -159,11 +175,28 @@ fn check_ty(cx: &LateContext, ast_ty: &Ty) {
                 },
             }
         },
+        TyRptr(_, MutTy { ref ty, .. }) => {
+            match ty.node {
+                TyPath(ref qpath) => {
+                    let def = cx.tables.qpath_def(qpath, ast_ty.id);
+                    if let Some(def_id) = opt_def_id(def) {
+                        if Some(def_id) == cx.tcx.lang_items.owned_box() {
+                            span_help_and_lint(cx,
+                                               BOX_VEC,
+                                               ast_ty.span,
+                                               "you seem to be trying to use `&Box<T>`. Consider using just `&T`",
+                                               "replace `&Box<T>` with simply `&T`");
```

Comment:
I just need to read through some other examples and I think I can use `span_suggestion`. Something like https://github.com/mcarton/rust-clippy/commit/44daa8bd72d29b281741a6fc11e07faa03d5fb1e?